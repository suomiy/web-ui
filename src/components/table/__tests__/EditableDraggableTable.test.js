import React from 'react';
import { shallow, mount } from 'enzyme';
import EditableDraggableTable from '../EditableDraggableTable';

const rows = [
  {
    id: 1,
    one: 'First Column',
    two: 'Second Column'
  },
  {
    id: 2,
    one: 'F Column',
    two: 'S Column'
  }
];

const columns = [
  {
    header: {
      label: 'One'
    },
    property: 'one'
  },
  {
    header: {
      label: 'Two'
    },
    property: 'two'
  }
];

const testEditableDraggableTable = (rowsArg = [], onChange = null) => (
  <EditableDraggableTable columns={columns} rows={rowsArg} onChange={onChange || jest.fn()} />
);

describe('<EditableDraggableTable />', () => {
  it('renders correctly', () => {
    const component = shallow(testEditableDraggableTable());
    expect(component).toMatchSnapshot();
  });

  it('renders data correctly', () => {
    const component = shallow(testEditableDraggableTable(rows));
    expect(component).toMatchSnapshot();
  });

  it('creates new row', () => {
    const onChange = jest.fn();
    const component = mount(testEditableDraggableTable(rows, onChange));

    component.setProps({
      onChange,
      rows: [
        ...rows,
        {
          id: 3,
          edit: true,
          one: 'F3 Column',
          two: 'S3 Column'
        }
      ]
    });
    expect(onChange).toHaveBeenCalled();
  });
});
