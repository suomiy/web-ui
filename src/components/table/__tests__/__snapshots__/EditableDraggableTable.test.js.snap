// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditableDraggableTable /> renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dragDropManager": DragDropManagerImpl {
        "backend": HTML5Backend {
          "actions": Object {
            "beginDrag": [Function],
            "drop": [Function],
            "endDrag": [Function],
            "hover": [Function],
            "publishDragSource": [Function],
          },
          "altKeyPressed": false,
          "asyncEndDragFrameId": null,
          "context": Object {},
          "currentDragSourceNode": null,
          "currentDragSourceNodeOffset": null,
          "currentDragSourceNodeOffsetChanged": false,
          "currentNativeHandle": null,
          "currentNativeSource": null,
          "dragEnterTargetIds": Array [],
          "dragOverTargetIds": null,
          "dragStartSourceIds": null,
          "dropTargetIds": Array [],
          "enterLeaveCounter": EnterLeaveCounter {
            "entered": Array [],
          },
          "monitor": DragDropMonitorImpl {
            "registry": HandlerRegistryImpl {
              "dragSources": Map {},
              "dropTargets": Map {},
              "pinnedSource": null,
              "pinnedSourceId": null,
              "store": Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              },
              "types": Map {},
            },
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "mouseMoveTimeoutTimer": null,
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "sourceNodeOptions": Map {},
          "sourceNodes": Map {},
          "sourcePreviewNodeOptions": Map {},
          "sourcePreviewNodes": Map {},
        },
        "context": Object {},
        "isSetUp": false,
        "monitor": DragDropMonitorImpl {
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
    }
  }
>
  <EditableDraggableTable
    columns={
      Array [
        Object {
          "header": Object {
            "label": "One",
          },
          "property": "one",
        },
        Object {
          "header": Object {
            "label": "Two",
          },
          "property": "two",
        },
      ]
    }
    onChange={[Function]}
    rows={Array []}
  />
</ContextProvider>
`;

exports[`<EditableDraggableTable /> renders data correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dragDropManager": DragDropManagerImpl {
        "backend": HTML5Backend {
          "actions": Object {
            "beginDrag": [Function],
            "drop": [Function],
            "endDrag": [Function],
            "hover": [Function],
            "publishDragSource": [Function],
          },
          "altKeyPressed": false,
          "asyncEndDragFrameId": null,
          "context": Object {},
          "currentDragSourceNode": null,
          "currentDragSourceNodeOffset": null,
          "currentDragSourceNodeOffsetChanged": false,
          "currentNativeHandle": null,
          "currentNativeSource": null,
          "dragEnterTargetIds": Array [],
          "dragOverTargetIds": null,
          "dragStartSourceIds": null,
          "dropTargetIds": Array [],
          "enterLeaveCounter": EnterLeaveCounter {
            "entered": Array [],
          },
          "monitor": DragDropMonitorImpl {
            "registry": HandlerRegistryImpl {
              "dragSources": Map {},
              "dropTargets": Map {},
              "pinnedSource": null,
              "pinnedSourceId": null,
              "store": Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              },
              "types": Map {},
            },
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
          },
          "mouseMoveTimeoutTimer": null,
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "sourceNodeOptions": Map {},
          "sourceNodes": Map {},
          "sourcePreviewNodeOptions": Map {},
          "sourcePreviewNodes": Map {},
        },
        "context": Object {},
        "isSetUp": false,
        "monitor": DragDropMonitorImpl {
          "registry": HandlerRegistryImpl {
            "dragSources": Map {},
            "dropTargets": Map {},
            "pinnedSource": null,
            "pinnedSourceId": null,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "types": Map {},
          },
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
    }
  }
>
  <EditableDraggableTable
    columns={
      Array [
        Object {
          "header": Object {
            "label": "One",
          },
          "property": "one",
        },
        Object {
          "header": Object {
            "label": "Two",
          },
          "property": "two",
        },
      ]
    }
    onChange={[Function]}
    rows={
      Array [
        Object {
          "id": 1,
          "one": "First Column",
          "two": "Second Column",
        },
        Object {
          "id": 2,
          "one": "F Column",
          "two": "S Column",
        },
      ]
    }
  />
</ContextProvider>
`;
